src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -83,27 +83,27 @@ float64_t RandomFourierDotFeatures::post_dot(float64_t dot_result, index_t par_i
 	return std::cos(dot_result) * constant;
 }
 
-SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
+void RandomFourierDotFeatures::generate_random_parameter_vector(SGMatrix<float64_t>& mat, index_t col)
```

Comment:
rather provide a result vector than a matrix

Reply:
Indeed, what you wrote he is something you’d see in C (the reference would be a pointer). I think this is google style C++, but we don’t use it, as in modern C++ the compiler optimises the return value allocation and we can even have multiple return statements.

Reply:
I don't think the compiler would optimise out the method allocating a new sgvector and the an memcpy call subsequently. Rather, we would allocate the matrix first, and then call this method here with wrapped column vectors (There is a method in SGMatrix that returns a non memory managed column vector)

Reply:
or would it @gf712 ? That would blow my mind :)

Reply:
No, what I mean is the sgmatrix allocation is done one the calling stack. So that it isn’t copied from this stack to the caller. The sgmatrix data pointer is in the heap, so that can’t be optimised here